src/components/tabs/CdrTabs.vue

```diff
@@ -89,276 +23,244 @@ export default defineComponent({
+  if (!previousIndexExists){
+      return;
+  }
+  selectTab(prevIndex)
+}
+
+const selectTab = async (index) => {
+  const tabToSelect = tabElements.value[index];  
+  selectedTabName.value = tabs.value[index].name;
+  selectedIndex.value = index;
+  await nextTick();
+  tabToSelect.focus();
+  updateUnderline();
+}
+
+onMounted(() => {
+    setInitialTabStates();
+    headerWidth.value = getHeaderWidth();
+    calculateOverflow();
+    setTimeout(()=>{
+        updateUnderline();
+    }, 250);
+    window.addEventListener('resize', debounce(() => {
         headerWidth.value = getHeaderWidth();
         calculateOverflow();
         updateUnderline();
-      }, 500));
-      // Check for header overflow on widow resize for gradient behavior.
-      cdrTabsHeaderEl.value.parentElement.addEventListener('scroll', debounce(() => {
+    }, 250));
+    tablist.value.addEventListener('scroll', debounce(() => {
         calculateOverflow();
         updateUnderline();
-      }, 50));
+    }, 50));
+})
+
+const setInitialTabStates = () => {
+    tabElements.value.forEach((tab, index) =>{
+        if(!tab.disabled && selectedIndex.value === null){
+            selectedIndex.value = index;
+            selectedTabName.value = tabs.value[index].name;
+        }
     })
+}
 
-    return {
-      style: useCssModule(),
-      mapClasses,
-      baseClass,
-      modifierClass,
-      sizeClass,
-      tabs,
-
-      // containerHeight, // ????
-      overflowLeft,
-      overflowRight,
-
-      rightArrowNav,
-      leftArrowNav,
-      handleDownArrowNav,
-      handleClick,
-
-      cdrTabsHeaderEl,
-      slotWrapperEl,
-      containerEl,
-
-      underlineStyle,
-      gradientLeftStyle,
-      gradientRightStyle,
-
-
-      getHeaderWidth,
-      calculateOverflow,
-      updateUnderline,
-      hideScrollBar,
-      changeTab,
-      getNextTab,
-      getPreviousTab,
-
-      activeTabIndex,
-      setFocusToActiveTabHeader,
-      headerOverflow,
-    };
-  },
-});
+const style = useCssModule();
 </script>
 
+<template>
+  <div
+    :class="mapClasses(style, baseClass, modifierClass, sizeClass)"
+    ref="containerEl"
+    :style="{ height }"
+  >
+    <div
+      :class="style['cdr-tabs__gradient-container']"
+    >
+      <div
+        :class="mapClasses(
+          style,
+          'cdr-tabs__gradient',
+          'cdr-tabs__gradient--left',
+          overflowLeft ? 'cdr-tabs__gradient--active' : ''
+        )"
+        :style="gradientLeftStyle"
+      />
+    <ul :class="style['cdr-tabs__header-container']" role="tablist" ref="tablist">
+      <li v-for="(tab, index) in tabs" 
+        :key="tab.name" role="presentation"
+        :class="style['cdr-tabs__header']" 
+      >
+        <button
+          :ref="el => { tabElements[index] = el }"
+          :id="getTabId(tab.name)"
+          :disabled="tab.disabled"
+          :aria-disabled="tab.disabled"
+          :aria-selected="(selectedIndex === index && !tab.disabled) ? true : false"
```

Comment:
Is there a way to get a computed value for `(selectedIndex === index && !tab.disabled)`? It's in a `v-for` loop so it didn't seem that simple. Anyway it bothered me that I was repeating this 3 times.